### ISSUE TITLE
CMake requirement is too high for Ubuntu Xenial (16.04)

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/Daemon#475]
During pre-0.52 development cycle I regularly tested building and running the game on Ubuntu Xenial (16.04), it worked fine.

That's why I written in [Dæmon 0.52 blog post](https://unvanquished.net/announcing-daemon-engine-beta-0-52/):

> We build the Unvanquished release on Debian Buster and the produced Linux build must work on distributions from the Ubuntu Bionic era (2018, we made specific efforts for this), and the engine is still buildable on Ubuntu Xenial (2016): our continuous integration system verifies it for us and we tested such binaries were playable.

But it looks like I missed something, I tried to build it today and got this:

```
cmake '/home/illwieckz/dev/buildme-unvanquished/Daemon' -B'/home/illwieckz/dev/buildme-unvanquished/build/engine/fast' \
	-DUSE_CURSES=OFF -D'USE_BREAKPAD=OFF' -D'CMAKE_BUILD_TYPE=Release' -D'USE_DEBUG_OPTIMIZE=OFF' -D'USE_LTO'='OFF' \
	 \
	-D'EXTERNAL_DEPS_DIR=/home/illwieckz/dev/buildme-unvanquished/build/deps' \
	-D'BUILD_SERVER=OFF' -D'BUILD_CLIENT=ON' -D'BUILD_TTY_CLIENT=OFF' \
	-G'Unix Makefiles'
CMake Error at CMakeLists.txt:27 (cmake_minimum_required):
  CMake 3.12 or higher is required.  You are running version 3.5.1
```

The requirement for 3.12 was added in f2fe511fdeebc34ab39a2e78edbd58eed371841e as part of #413 to fix #411.

Due to the proprietary nature of Nvidia drivers, some old Nvidia GPUs may require Xenial, that's why I tested Xenial before releasing (and those GPUs doesn't have featureful enough free open source drivers). Though, if they are supposedly able to run the game in term of supported features, they are old and slow.

**Comment by slipher** [comment]
The Xenial environment on Travis is still working. It supplies CMake 3.12: https://docs.travis-ci.com/user/reference/xenial/#compilers-and-build-toolchain

**Comment by illwieckz** [comment]
Yes, that's what makes me wrongly believe the engine was still building on Xenial, but cmake 3.12 is not provided by Xenial itself:

```
illwieckz@test:~$ apt-cache show cmake | grep ^Version
Version: 3.5.1-1ubuntu3
Version: 3.5.1-1ubuntu1
illwieckz@test:~$ apt-cache show cmake | grep '^Version'
Version: 3.5.1-1ubuntu3
Version: 3.5.1-1ubuntu1
```

Note: I haven't tried `xenial-backports`, but those are expected to be disabled by default.

**Comment by slipher** [comment]
Xenial already reached end of support on April 30, 2021. If there is a CMake backport available, that seems good enough to me; the CMake ugliness to support old versions doesn't seem worth it.

**Comment by illwieckz** [comment]
Hmm, I now notice I had backports enabled, so I confirm there is no cmake 3.12 in Xenial backport.

Kitware provides a repository for Xenial cmake packages (currently from 3.15 to 3.20): https://apt.kitware.com/

But that's not an official Xenial repository.

**Comment by illwieckz** [comment]
According to https://wiki.ubuntu.com/Releases Xenial reached _End of Standard Support_ in April 2021 and will reach _End of Life_ in April 2024. According to https://ubuntu.com/about/release-cycle and https://wiki.ubuntu.com/ReleaseSchedule it has entered _Extended Security Maintenance_.

**Comment by slipher** [comment]
> Customers of Canonical often ask for an extended security maintenance commitment, either to ‘main’ for a longer period of time, or to ‘universe’ software packages during the initial maintenance period of the LTS. This is known as ‘Extended Security Maintenance’ or ESM and is available for LTS releases since Ubuntu 12.04 LTS.

>  It is recommended for users and organisations to upgrade to the latest LTS release or subscribe to ESM for continued security coverage.

It is only relevant for paid customers

**Comment by illwieckz** [comment]
For reference I managed to build the Dæmon engine 0.52 on Ubuntu Trusty (14.04) with:

- CMake 3.12 installed with https://cmake.org/files/v3.12/cmake-3.12.0-Linux-x86_64.sh
- clang 5.0 installed with `deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-5.0 main`

and run the game and join a server successfully (this is lowest preset on software rendering because of virtual machine):

[![Unvanquished 0.52 on Ubuntu Trusty](https://dl.illwieckz.net/b/daemon/features/system-support/20210602-060440-000.unvanquished-0.52-ubuntu-14.04-amd64.png)](https://dl.illwieckz.net/b/daemon/features/system-support/20210602-060440-000.unvanquished-0.52-ubuntu-14.04-amd64.png)

So, basically, there is solid solutions to build the game on old systems, and multiple working solutions for CMake itself.

Maybe we can just focus on CMake 3.12 then…

Side note: I had to install CMake 3.12 on Trusty even with my commits bringing back older CMake despite the `CMakeLists.txt` file support only requiring 2.8.12 (which is provided by Trusty, that's why I tried Trusty) because the `FindLua.cmake` uses features from higher version of CMake and I don't know which version is required by them…

**Comment by illwieckz** [comment]
Let's consider this fixed.
